presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java

```diff
@@ -74,9 +89,29 @@ protected Void visitLogicalBinaryExpression(LogicalBinaryExpression node, Materi
         if (!node.getOperator().equals(LogicalBinaryExpression.Operator.AND)) {
             throw new SemanticException(NOT_SUPPORTED, node, "Only AND operator is supported for join criteria for materialized view.");
         }
+
+        visitLogicalBinaryExpressionChild(node.getLeft(), context);
+        visitLogicalBinaryExpressionChild(node.getRight(), context);
+
```

Comment:
Good, this solves the recursive problem. Please remember we still need to check if "=" exists in the leaf node.

Reply:
Yes, Lin is correct that we do want to check the leaf node is "=" ComparisonExpression but nothing else. The standard way should be to implement the ComparisonExpression interface, and check the Operator.EQUAL there.

Separately, if you look into the base class default implementation of super.visitLogicalBinaryExpression(node, context), you'll find you don't need the above two lines of visitLogicalBinaryExpressionChild. Calling super.visitLogicalBinaryExpression(node, context) is sufficient. Sorry, we weren't very clear in the Issue description.

> 

@Override
protected R visitLogicalBinaryExpression(LogicalBinaryExpression node, C context)
{
    process(node.getLeft(), context);
    process(node.getRight(), context);

    return null;
}
If the node.getLeft() is a LogicalBinaryExpression, process(node.getLeft()) will call visitLogicalBinaryExpression() on node.getLeft() recursively.

If you follow more on process(Node node), you will see it eventually calls visitT() where T is the actual type/class of node. This idea, as well as the recursion idea Lin mentioned, is very common for AST visitor implementation, in general.